sympy/integrals/transforms.py

```diff
@@ -1169,6 +1734,59 @@ def _collapse_extra(self, extra):
                 'Laplace', None, 'No combined convergence.')
         return plane, cond
 
+    def doit(self, **hints):
+        """
+        Try to evaluate the Laplace transform in closed form.
+        """
+
+        needeval = hints.pop('needeval', False)
+        fn = self.function
+        t_ = self.function_variable
+        s_ = self.transform_variable
+
+        if not fn.is_Add:
+            fn = expand_mul(fn)
+
+        if not fn.is_Add:
+            try:
+                return self._compute_transform(fn, t_, s_, **hints)
+            except IntegralTransformError:
+                pass
+        else:
+            hints['needeval'] = needeval
+            res = [self.__class__(*([x] + list(self.args[1:]))).doit(**hints)
+                   for x in fn.args]
+            extra = []
+            ress = []
+            for x in res:
+                if not isinstance(x, tuple):
+                    x = [x]
+                ress.append(x[0])
+                if len(x) == 2:
+                    # only a condition
+                    extra.append(x[1])
+                elif len(x) > 2:
+                    # some region parameters and a condition (Mellin, Laplace)
+                    extra += [x[1:]]
```

Comment:
extra.append? (Compare three rows up. Or change there?)

Reply:
Hmm, I have no clue ;)  `LaplaceTransform` did not have its own `doit` before, so this part of the code is a verbatim copy of the `doit` of `IntegralTransform`

https://github.com/sympy/sympy/blob/b5191b805eba30488a54cc0e46dbab66052a0429/sympy/integrals/transforms.py#L175-L180

If we change this in one place, we should also cgange it in the other place.

Reply:
It's better to avoid duplicating the same code in multiple places. Can one method not just call another or can there be a third method that they both call or something?

Reply:
This whole block is duplicated:

https://github.com/sympy/sympy/blob/b5191b805eba30488a54cc0e46dbab66052a0429/sympy/integrals/transforms.py#L165-L201

So I could convert it to a new function inside `IntegralTransform`. Is that a proper way to solve it?

Reply:
There are many possible ways to do this without duplication. The question is which part of the code needs to be *different* in the subclass from that of the superclass? That part of the code can be moved into a separate method that is called by the superclass `doit` method. Then the subclass can override that method and have it do something different while both classes can share the code that needs to be the *same* for both classes.

Reply:
We can discuss it here: https://github.com/sympy/sympy/pull/22376#discussion_r761809377

If that one is solved, this one is automatically solved as well.